File: backend/app/models/order.py
from sqlalchemy import Column, Integer, ForeignKey, Numeric, String, DateTime
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.db.base import Base


class Order(Base):
    __tablename__ = "orders"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)

    total_amount = Column(Numeric(10, 2), nullable=False)
    status = Column(String(20), default="PENDING")
    tracking_number = Column(String(50), unique=True, index=True)

    created_at = Column(DateTime(timezone=True), server_default=func.now())

    user = relationship("User", back_populates="orders")

    items = relationship(
        "OrderItem",
        back_populates="order",
        cascade="all, delete-orphan",
    )

File: frontend/lib/models/order.dart
class Order {
  final int id;
  final String status;
  final String? trackingNumber;
  final String totalAmount;
  final DateTime createdAt;
  

  Order({
    required this.id,
    required this.status,
    this.trackingNumber,
    required this.totalAmount,
    required this.createdAt,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['id'],
      status: json['status'],
      trackingNumber: json['tracking_number'],
      totalAmount: json['total_amount'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }
}
